src/mscorlib/corefx/System/Globalization/DateTimeFormatInfo.cs

```diff
@@ -2808,6 +2801,31 @@ void InsertHash(TokenHashValue[] hashTable, String str, TokenType tokenType, int
             } while (i < TOKEN_HASH_SIZE);
             Contract.Assert(true, "The hashtable is full.  This should not happen.");
         }
+
+        private bool CompareStringOptimized(string string1, int offset1, int length1, string string2, int offset2, int length2)
+        {
+            if (length1 != length2)
```

Comment:
if we really need to optimize for 1 character string length cases, we need to do it explicitly like 

if (length1==1 && length2==1 && string1[offset1] == string2[offset2])
 // do whatever 
else
// do the cultural comparison 

it is not worth to call casing here because the common cases you are trying to optimize will be characters not needing casing.